src/actions/NamespaceAction.ts

```diff
@@ -0,0 +1,64 @@
+import * as Api from '../services/Api';
+
+export enum NamespaceActionType {
+  RELOAD = 'reload',
+  API_INITIATE_REQUEST = 'api_initiate_request',
+  API_RECEIVE_LIST = 'api_receive_list'
+}
+
+export const reload = () => {
+  return {
+    type: NamespaceActionType.RELOAD
+  };
+};
+
+export const apiInitiateRequest = () => {
+  return {
+    type: NamespaceActionType.API_INITIATE_REQUEST
+  };
+};
+
+export const apiReceiveList = newList => {
+  return {
+    type: NamespaceActionType.API_RECEIVE_LIST,
+    list: newList,
+    receivedAt: Date.now()
+  };
+};
+
+export const asyncFetchNamespaces = () => {
+  return dispatch => {
+    dispatch(apiInitiateRequest());
+    return Api.GetNamespaces()
+      .then(response => response['data'])
+      .then(data => dispatch(apiReceiveList(data)));
+  };
+};
+
+const shouldFetchNamespaces = state => {
```

Comment:
what  happens if a new namespace is added in OpenShift/k8s after the initial namespace fetch is done?  Am I reading this right or do the namespaces only get fetched once and never more? Thus if a new namespace is added after, it will never be seen.

Reply:
whenever the user clicks on the dropdown the namespace will get reloaded from the backend.  In the future we can have a separate timer-driven refreshing logic.